src/python/pants/build_graph/build_configuration.py

```diff
@@ -17,23 +19,23 @@
 logger = logging.getLogger(__name__)
 
 
+@dataclass
 class BuildConfiguration:
     """Stores the types and helper functions exposed to BUILD files."""
 
+    _target_by_alias: Dict[Any, Any] = field(default_factory=dict)
+    _target_macro_factory_by_alias: Dict[Any, Any] = field(default_factory=dict)
+    _exposed_object_by_alias: Dict[Any, Any] = field(default_factory=dict)
+    _exposed_context_aware_object_factory_by_alias: Dict[Any, Any] = field(default_factory=dict)
+    _optionables: OrderedSet = OrderedSet()
+    _rules: OrderedSet = OrderedSet()
```

Comment:
Do you now what these are `OrderedSet`s of? It's okay if we don't - this is still an improvement, but would be even better with the types specified.

Reply:
Afraid not, I just added this to make mypy stop complaining.